dts/bindings/adc/adc-controller.yaml

```diff
@@ -12,3 +13,129 @@ properties:
     "#io-channel-cells":
       type: int
       required: true
+
+    "#address-cells":
+      required: false
+      const: 1
+
+    "#size-cells":
+      required: false
+      const: 0
+
+child-binding:
+    description: |
+      Channel configuration.
+
+      All nodes using this binding must be named "channel", otherwise their
+      data will not be accessible for the ADC API macros.
+
+    properties:
+      reg:
+        type: array
+        required: true
+        description: Channel identifier.
+
+      gain:
+        type: string
+        required: true
+        description: |
+          Gain selection:
+          - ADC_GAIN_1_6: x 1/6
+          - ADC_GAIN_1_5: x 1/5
+          - ADC_GAIN_1_4: x 1/4
+          - ADC_GAIN_1_3: x 1/3
+          - ADC_GAIN_1_2: x 1/2
+          - ADC_GAIN_2_3: x 2/3
+          - ADC_GAIN_1:   x 1
+          - ADC_GAIN_2:   x 2
+          - ADC_GAIN_3:   x 3
+          - ADC_GAIN_4:   x 4
+          - ADC_GAIN_6:   x 6
+          - ADC_GAIN_8:   x 8
+          - ADC_GAIN_12:  x 12
+          - ADC_GAIN_16:  x 16
+          - ADC_GAIN_24:  x 24
+          - ADC_GAIN_32:  x 32
+          - ADC_GAIN_64:  x 64
+          - ADC_GAIN_128: x 128
+        enum:
+          - "ADC_GAIN_1_6"
+          - "ADC_GAIN_1_5"
+          - "ADC_GAIN_1_4"
+          - "ADC_GAIN_1_3"
+          - "ADC_GAIN_1_2"
+          - "ADC_GAIN_2_3"
+          - "ADC_GAIN_1"
+          - "ADC_GAIN_2"
+          - "ADC_GAIN_3"
+          - "ADC_GAIN_4"
+          - "ADC_GAIN_6"
+          - "ADC_GAIN_8"
+          - "ADC_GAIN_12"
+          - "ADC_GAIN_16"
+          - "ADC_GAIN_24"
+          - "ADC_GAIN_32"
+          - "ADC_GAIN_64"
+          - "ADC_GAIN_128"
+
+      reference:
+        type: string
+        required: true
+        description: |
+          Reference selection:
+          - ADC_REF_VDD_1:     VDD
+          - ADC_REF_VDD_1_2:   VDD/2
+          - ADC_REF_VDD_1_3:   VDD/3
+          - ADC_REF_VDD_1_4:   VDD/4
+          - ADC_REF_INTERNAL:  Internal
+          - ADC_REF_EXTERNAL0: External, input 0
+          - ADC_REF_EXTERNAL1: External, input 1
+        enum:
+          - "ADC_REF_VDD_1"
+          - "ADC_REF_VDD_1_2"
+          - "ADC_REF_VDD_1_3"
+          - "ADC_REF_VDD_1_4"
+          - "ADC_REF_INTERNAL"
+          - "ADC_REF_EXTERNAL0"
+          - "ADC_REF_EXTERNAL1"
+
+      vref-mv:
+        type: int
+        required: false
+        description: |
+          This property can be used to specify the voltage (in millivolts)
+          of the reference selected for this channel, so that applications
+          can get that value if needed for some calculations.
+          For the internal reference, the voltage can be usually obtained with
+          a dedicated ADC API call, so there is no need to use this property
+          in that case, but for other references this property can be useful.
```

Comment:
For example, this description is referencing the ADC API, indicating that something zephyr-specific may be going on here.